subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt

```diff
@@ -17,37 +17,97 @@
 #
 
 ##############################################################################
-##
-##  ${applicationName} start up script for UN*X
-##
+#
+#   Gradle start up script for POSIX
+#
+#   You're probably reading this because you're a Unix/Linux person and have
+#   been instructed to run "gradlew" to install or build a project that's of
+#   interest to you. (At least, that's how I wound up reading this script.)
+#
+#   Here's some stuff you need to know.
+#
+#   (0) You need a POSIX-compliant shell to run this script. If your /bin/sh
+#       is noncompliant, but you have some other shell that is (perhaps ksh or
+#       bash) then to run this script, type the shell name before the whole
+#       command line, like:
+#
+#           ksh gradlew [...args...]
+#
+#   (1) Even if you have a compliant shell, Java is fussy about $PWD, so you
+#       might as well cd to the directory containing gradlew and run it as
+#
+#           ./gradlew
+#
+#   (2) Gradle is a build system used by the project that you're building or
+#       installing. It's like "autoconf+make" but for projects that are written
+#       in Java.
+#
+#   (3) Gradle is mostly written in Java, but there's wrapper scripts for
+#       various environments, including this one. You can find it at
+#       https://github.com/gradle/gradle
+#
+#   (4) Because we're targeting any POSIX shell, we progressively modify "$@",
+#       rather than try and built up all the args just before invoking Java.
+#       We even pass shellcheck.
+#
+#   (5) This script (gradlew) is generated from
+#       subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt
+#       in the gradle project, so submitting patches will be tedious.
+#<%
+#       If you're reading this, this IS the template just mentioned.
+#
+#   (6) There are code segments with < % ... % > that control conditional
+#       parts. The two main cases are:
+#           (a) to build a module rather than an app, indicated by passing '--module XYZ'
+#           (b) if appNameSystemProperty is set, include a -D definition for it
+#
+#   (7) Every $ and \ has to be escaped with a \
+#
+#   (8) Any unescaped $ introduces a template parameter to be substituted.
+#
+#    (a) a bunch of boilerplate (if these ever change, submit a PR for this file)
+#
+#       ${appNameSystemProperty}    org.gradle.appname
+#       ${applicationName}          Gradle
+#       ${mainClassName}            org.gradle.wrapper.GradleWrapperMain
+#       ${optsEnvironmentVar}       GRADLE_OPTS
+#
+#    (b) some info about your build settings:
+#
+#       ${defaultJvmOpts}           varies, but here it's:
+#                                   -Dfile.encoding=UTF-8 "-Xmx64m" "-Xms64m"
+%>
+#
 ##############################################################################
```

Comment:
To try this locally, I ended up just deleting this entire comment block. I think the unescaped `$` in here breaks the generator.

Reply:
That was my unfamiliarity with Groovy; the comment needs `/*...*/` rather than `#`, so I'm adjusting that.